language/move-lang/src/parser/syntax.rs

```diff
@@ -1453,34 +1486,22 @@ fn parse_module<'input>(tokens: &mut Lexer<'input>) -> Result<ModuleDefinition,
     let name = parse_module_name(tokens)?;
     consume_token(tokens, Tok::LBrace)?;
 
-    let mut uses = vec![];
-    while tokens.peek() == Tok::Use {
-        uses.push(parse_use_decl(tokens)?);
-    }
-
-    let mut structs = vec![];
-    let mut functions = vec![];
-    let mut specs = vec![];
+    let mut members = vec![];
     while tokens.peek() != Tok::RBrace {
-        if tokens.peek() == Tok::Spec {
-            specs.push(parse_spec_block(tokens)?);
-        } else if is_struct_definition(tokens)? {
-            structs.push(parse_struct_definition(tokens)?);
-        } else {
-            functions.push(parse_function_decl(tokens, /* allow_native */ true)?);
-        }
+        members.push(match tokens.peek() {
+            Tok::Spec => ModuleMember::Spec(parse_spec_block(tokens)?),
+            Tok::Use => ModuleMember::Use(parse_use_decl(tokens)?),
+            // TODO rework parsing modifiers
+            _ if is_struct_definition(tokens)? => {
+                ModuleMember::Struct(parse_struct_definition(tokens)?)
+            }
+            _ => ModuleMember::Function(parse_function_decl(tokens, /* allow_native */ true)?),
+        })
     }
-    tokens.advance()?; // consume the RBrace
+    consume_token(tokens, Tok::RBrace)?;
```

Comment:
Why change this? It's a redundant check, isn't it? You can only exit the preceding loop on an RBrace token. Maybe I was thinking too much about performance, but this used to be a pretty common idiom in the parser. I just looked again and it isn't used so much now. I don't really care either way, I guess.

Reply:
I guess I really wanted an assert here? something like `assert!(tokens.advance()? == Tok::RBrace)`?

Do you have anything in place that behaves that way?

Reply:
consume_token is as good as anything for that.